ai/core/src/unitycatalog/ai/core/utils/validation_utils.py

```diff
@@ -140,3 +140,55 @@ def validate_function_name_length(function_name: str) -> None:
             f"The maximum length of a function name is {OSS_MAX_FUNCTION_NAME_LENGTH}. "
             f"The name supplied is {name_length} characters long."
         )
+
+
+def is_valid_retriever_output(output: Any) -> bool:
+    """
+    Checks if the given output follows the retriever format for MLflow, which is a list of Documents
+    or dictionaries that follow the Document format.
+
+    Args:
+        output: The value to determine if it is a valid retriever output.
+
+    Returns:
+        bool: If the provided output is a valid retriever output.
+    """
+    if not isinstance(output, list):
+        return False
+
+    def is_valid_retriever_item(item: Any) -> bool:
+        from mlflow.entities import Document
+
+        if isinstance(item, Document):
+            return True
+
+        if isinstance(item, dict):
+            try:
+                _ = Document(**item)
+                return True
+            except TypeError:
+                return False
+
+        return False
+
+    return all(is_valid_retriever_item(item) for item in output)
+
+
+def autologging_is_enabled(integration_name):
```

Comment:
Could we rename to something like `mlflow_tracing_enabled`? (since there're flavors that don't support tracing in autologging) And we should check if `log_traces` is set to True or not right, though by default it's True.